crates/consensus/authority/src/healthcheck_task.rs

```diff
@@ -0,0 +1,257 @@
+use std::{
+    collections::{HashMap, HashSet},
+    sync::Arc,
+    time::Instant,
+};
+
+use crate::Storage;
+use reth_network::{
+    frost::{
+        manager::{FrostCommand, ToFrostManager},
+        PeerMessageResponse,
+    },
+    NetworkHandle,
+};
+use reth_network_api::Peers;
+use reth_rpc_types::PeerId;
+use reth_tasks::TaskExecutor;
+use tokio::sync::RwLock;
+use tracing::{error, info, warn};
+
+const NONRESPONDING_PEERS_TIMEOUT_SECS: u64 = 45;
+
+pub struct HealthcheckTask<ToFrostMan> {
+    /// Network Handler
+    pub(crate) network_handle: NetworkHandle,
+    /// Frost network Handler
+    pub(crate) frost_handle: ToFrostMan,
+    /// Shared storage to insert aggregate public key
```

Comment:
This task doesnt insert keys right?
```suggestion
    /// Shared authority storage 
```
You can also not use the AuthorityStorage trait

Reply:
changed, thx